File: AVL/AVL-Csharp/src/Node.cs
using System;

namespace src
{
  struct Node 
  {
    Node(Data data)
    {
      this.right=null;
      this.left=null;
      this.height=1;
      this.data=data;
    }

    public Data data{get;}
    public Node right{get;set;}
    public Node left{get;set;}
    public int height{get;set;}
  }
}

File: AVL/DART/Node.dart
import 'Data.dart';

class Node 
{
  Node(this._data);

  get rigth=>this._right;
  get left=>this._left;
  get data=>this._data;
  get height=>this._height;
  set height(int n){this._height=n;}
  set data(Data data){this._data=data;}
  set right(Node node){this._right=node;}
  set left(Node node){this._left=node;}

  Data _data;
  Node? _right,_left;
  int _height=1;
}
